lib/cfn-nag/cfn_nag_executor.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+require 'trollop'
+require 'cfn-nag/cli_options'
+require 'cfn-nag/cfn_nag_config'
+
+class CfnNagExecutor
+  def initialize(require_input_path: false)
+    @require_input_path = require_input_path
+    @profile_definition = nil
+    @blacklist_definition = nil
+    @parameter_values_string = nil
+  end
+
+  def scan(aggregate_output: true,
+           cli_supplier: cli_options(cfn_nag_scan: @require_input_path),
```

Comment:
if i follow the code, cli_supplier and arg_supplier are exposed here to improve testability?  imho this is a tradeoff for complicating the interface - i.e. anybody who calls this might scratch their head and likely never change the defaults.  another approach i can suggest is to have "factory methods" for these two arguments, and then in your testing do a partial mock to replace the factory methods.  easy on the testing side, and nothing exposed to the consumer